### PR TITLE
feat: api requests

### LINKED ISSUE #36: feat: utils (front-end)
**Task**
What is the task? Adding api requests in the utils folder.

**More about the task (optional)**
This will happen client-side/front-end.
I'll use Axios for this.

**Alternatives you've considered (optional)**

**Additional context (optional)**

### PR DESCRIPTION
**Pull request by MaestroDagan**
closes #36 

Made functions in the "services" folder that make all the needed api requests.

## Summary
- installed Axios.
- removed .gitkeep.
- created services/index.js.
## Checklist
- [ ] fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] tested locally.
- [x] No error nor warning in the console.

### CODE REVIEW

**Comment by cherylli** [comment]
> closes #36
> 
> Made functions in the "services" folder that make all the needed api requests.
> 
> ## Summary
> * installed Axios.
> * removed .gitkeep.
> * created services/index.js.
> 
> ## Checklist
> * [ ]  fixed|updated|added unit tests and integration tests for each feature (if applicable).
> * [ ]  tested locally.
> * [x]  No error nor warning in the console.

does that mean it's not tested locally?

**Comment by MaestroDagan** [comment]
Yes, since I'm not really a front-end developer i didn't really know where I'd place it so it'll test it, but i could try to run some tests tho.

**Comment by MaestroDagan** [comment]
Okay, I'll test it rn

**Comment by MaestroDagan** [comment]
Everything works!

File: client/src/Components/UrlShortening/UrlShortening.jsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import classes from "./UrlShortening.module.css";
 function UrlShortening() {
   const urlInputRef = useRef();
 
-  function submitHandler(event) {
+  async function submitHandler(event) {
     event.preventDefault();
     const enteredUrl = urlInputRef.current.value;
     console.log(enteredUrl);
```
[the remark this answers is not in the document]
Comment (MaestroDagan) [reply client/src/Components/UrlShortening/UrlShortening.jsx:10]: > why do you have to change all localhost to 127.0.0.1? localhost should work. Your submit handler is not calling the axios stuff How did you test it?

I deleted all the testing, i tested everyting in the submitHandler. And I don't really know why I did that, I really should just start to add comments in moments like that

[the remark this answers is not in the document]
Comment (MaestroDagan) [reply client/src/Components/UrlShortening/UrlShortening.jsx:10]: > please remove console.log for development

It was already in there. I didn't place it.

File: client/src/services/index.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
-import { axios } from 'axios';
+import axios  from 'axios';
 
-const apiURL = 'http://localhost:4002'
+const apiURL = 'http://127.0.0.1:4002/api/links'
```
[the remark this answers is not in the document]
Comment (MaestroDagan) [reply client/src/services/index.js:3]: yeh, I could add that

File: client/src/Components/UrlShortening/UrlShortening.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaestroDagan) [reply client/src/Components/UrlShortening/UrlShortening.jsx:10]: I could make it the createLink function though, yeh i think that's handy. Do you think that's a good idea? @cherylli

[the remark this answers is not in the document]
Comment (MaestroDagan) [reply client/src/Components/UrlShortening/UrlShortening.jsx]: I did it!

File: client/src/services/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaestroDagan) [reply client/src/services/index.js]: its that way now

[the remark this answers is not in the document]
Comment (MaestroDagan) [reply client/src/services/index.js]: and its just localhost

[the remark this answers is not in the document]
Comment (cherylli) [reply client/src/services/index.js]: I know it's localhost, why change it?

**Comment by cherylli** [comment]
Mate, i don't think you've tested it. It has many errors and not running on my machine. index page is blank and getting error in console.

**Comment by MaestroDagan** [comment]
> hello, i don't think you've tested it. It has many errors and not running on my machine. index page is blank and getting error in console.

That's really weird, shall i send proof later today that it works perfectly fine on my machine?

**Comment by cherylli** [comment]
It needs to be working on every machine. Maybe someone else can check it out if they are free.

I see some syntax errors so I don't expect it to be fully working.

**Comment by klezi10** [comment]
@cherylli I'll take a look

**Comment by klezi10** [comment]
I'm getting a blank screen as well with this error @MaestroDagan 

image

**Comment by MaestroDagan** [comment]
Maybe it's a good idea for me to test codes on multiple devices

**Comment by MaestroDagan** [comment]
oh, that's weird

**Comment by MaestroDagan** [comment]
I'll take a quick look

**Comment by cherylli** [comment]
I'm interested to know what setup you have for this code to work

**Comment by MaestroDagan** [comment]
> 

Maybe, it's because i fixed the conflicts in the yarn file, i'm going to reinstall everything rn

**Comment by MaestroDagan** [comment]
Oh, I didn't test after i added dotenv, that's the problem

**Comment by cherylli** [comment]
You don't need to use dotenv for react. Also you should always check everything is working after you added something.

**Comment by MaestroDagan** [comment]
> You don't need to use dotenv for react. Also you should always check everything is working after you added something.

yes, I'll do that. Sorry for all the problems i bring with me.

**Comment by MaestroDagan** [comment]
Done, and i tested it!

**Comment by MaestroDagan** [comment]
How are yall doing?

**Comment by cherylli** [comment]
Hi please make the follow changes
- rename index to links.js 
- put baseURL in .env
- edit .env.template
- add await to createLink
also before you push, 
1. pull all the latest changes from main
2. resolve all the conflicts and 
3. test everything again

**Comment by MaestroDagan** [comment]
> * resolve all the conflicts and

Okay, I'll do that, but I can't use .env, because it's in the front-end

**Comment by cherylli** [comment]
hi are you still working on this?

**Comment by cherylli** [comment]
close - inactive/unresolved conflicts

**Pull request closed without merging**